src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -253,46 +255,155 @@ def _get_inferred_asset_deps(
             )
             maybe_disambiguated = explicitly_provided_deps.disambiguated(possible_addresses)
             if maybe_disambiguated:
-                yield maybe_disambiguated
+                return ImportResolveResult(ImportOwnerStatus.disambiguated, (maybe_disambiguated,))
+            else:
+                return ImportResolveResult(ImportOwnerStatus.ambiguous)
+        else:
+            return ImportResolveResult(ImportOwnerStatus.unowned)
+
+    return {filepath: _resolve_single_asset(filepath) for filepath in assets}
+
+
+class ImportOwnerStatus(Enum):
+    unambiguous = "unambiguous"
+    disambiguated = "disambiguated"
+    ambiguous = "ambiguous"
+    unowned = "unowned"
+    weak_ignore = "weak_ignore"
+    unownable = "unownable"
+
+
+@dataclass(frozen=True)
+class ImportResolveResult:
```

Comment:
Thoughts on whether I should add smart constructors? It's would help ensure that we always have the expected field combinations (ex every unambiguous result has an owner), but also it's more boilerplate.
ex:
```python
@staticmethod
def disambiguated(maybe_disambiguated):
    return ImportResolveResult(ImportOwnerStatus.disambiguated, (maybe_disambiguated,))
```

Reply:
Fine either way.

Reply:
I think I'm going to skip it, save's on a layer of indirection